### ISSUE TITLE
Upstreaming useful functionality to Perfetto UI

### ISSUE THREAD

**Issue by LalitMaganti** [janestreet/perfetto#22]
Hi folks,

I'm part of the Perfetto team and really love your tools; I've started playing around with it to actually optimize the performance of trace processor itself (using tools to make the tools themselves better; one of the benefits to working on performance!)!

I see from this repo that there are a few changes which are modifying the Perfetto UI in ways which would be useful to have upstream. Specifically, I'm thinking the following changes:
* https://github.com/janestreet/perfetto/commit/b8aef9cbdb27fd22d604703e1fe41444d975f96b
* https://github.com/janestreet/perfetto/commit/fb6cc517b39f24ba5f273125993358e93664def4
* https://github.com/janestreet/perfetto/commit/1f8063596b92e08fe5c16cc35d84a4571ab8764e
* https://github.com/janestreet/perfetto/commit/87abbd607d68e7e9aea3e95b4c182e4fabd2cb30
* https://github.com/janestreet/perfetto/commit/1b4765d74d78a65e492d820b8ba8de2c22a93e8d
* https://github.com/janestreet/perfetto/commit/323c1091f7f91851f103494e377fa15724dfa1df
* https://github.com/janestreet/perfetto/commit/64908117372bf60e4ae6f351804c439ccdb8da8f
* https://github.com/janestreet/perfetto/commit/4b68f1fe0e94f79562488dfa3c27ccaf49133e94
* https://github.com/janestreet/perfetto/commit/c1ce532aa1c99a1353a3b1f37cf272d643c43836

Some of these, I know Tristan sent our way before he left and are now in limbo; would be great if a magic-trace maintainer could take this over: we'd be very happy to have them!

**Comment by cgaebel** [comment]
Absolutely. Tudor's going to work on getting those upstreamed. Sorry we've been sitting on them for so long.
